mono/metadata/class.c

```diff
@@ -5808,6 +5817,7 @@ mono_class_create_from_typedef (MonoImage *image, guint32 type_token, MonoError
 	if (klass->generic_container) {
 		klass->is_generic = 1;
 		klass->generic_container->owner.klass = klass;
+		klass->generic_container->is_anonymous = FALSE; // Owner class is now known, container is no longer anonymous
```

Comment:
I'm extending the comments on `is_anonymous` to clarify, but: `mono_metadata_load_generic_params` cannot set the owner because it does not know who the owner is, the owner is always assigned by `mono_metadata_load_generic_params`'s caller. Until the owner is known, `is_anonymous` must be TRUE. In other words, `is_anonymous` is not "initialized" here; it changes.

It would be possible for `mono_metadata_load_generic_params` to take some representation of the owner as an argument. If we did this, we would remove the quirk where some containers are "temporarily" anonymous during initialization. Would you prefer this approach?